src/ports/postgres/modules/utilities/minibatch_preprocessing.py_in

```diff
@@ -0,0 +1,559 @@
+        """.format(
+        output_standardization_table = self.output_standardization_table,
+        grouping_cols = self.grouping_cols,
+        x_mean_str = self.x_mean_str,
+        x_std_dev_str = self.x_std_dev_str)
+        plpy.execute(query)
+
+class MiniBatchSummarizer:
+    @staticmethod
+    def create_output_summary_table(source_table, output_table,
+                                    dep_var_array_str, indep_var_array_str,
+                                    buffer_size, class_values, num_rows_processed,
+                                    num_missing_rows_skipped, output_summary_table):
+        query = """
+            CREATE TABLE {output_summary_table} AS
+            SELECT '{source_table}'::TEXT AS source_table,
+            '{output_table}'::TEXT AS output_table,
+            '{dependent_varname}'::TEXT AS dependent_varname,
+            '{independent_varname}'::TEXT AS independent_varname,
+            {buffer_size} AS buffer_size,
+            {class_values} AS class_values,
+            {num_rows_processed} AS num_rows_processed,
+            {num_missing_rows_skipped} AS num_missing_rows_skipped,
+            {grouping_cols}::TEXT AS grouping_cols
+        """.format(output_summary_table = output_summary_table,
+                   source_table = source_table,
+                   output_table = output_table,
+                   dependent_varname = dep_var_array_str,
+                   independent_varname = indep_var_array_str,
+                   buffer_size = buffer_size,
+                   class_values = class_values,
+                   num_rows_processed = num_rows_processed,
+                   num_missing_rows_skipped = num_missing_rows_skipped,
+                   grouping_cols = "NULL")
+        plpy.execute(query)
+
+class MiniBatchBufferSizeCalculator:
+    """
+    This class is responsible for calculating the buffer size.
+    This is a work in progress, final formula might change.
+    """
+    @staticmethod
+    def calculate_default_buffer_size(buffer_size,
+                                      num_rows_processed, independent_var_dimension):
+        if buffer_size is not None:
+            return buffer_size
+        num_of_segments = get_seg_number()
+
+        default_buffer_size = min(75000000.0/independent_var_dimension,
+                                    float(num_rows_processed)/num_of_segments)
+        return ceil(default_buffer_size)
+
+class MiniBatchDocumentation:
+    @staticmethod
+    def minibatch_preprocessor_help(schema_madlib, message):
+        method = "minibatch_preprocessor"
+        summary = """
+        ----------------------------------------------------------------
+                            SUMMARY
+        ----------------------------------------------------------------
+        MiniBatch Preprocessor is a utility function to pre process the input
+        data for use with models that support mini-batching as an optimization
+
+        #TODO add more here
```

Comment:
Is the comment still required?

Reply:
Yes. This is a placeholder for adding more detail description later.